nerfstudio/models/splatfacto.py

```diff
@@ -770,21 +778,41 @@ def get_outputs(self, camera: Cameras) -> Dict[str, Union[torch.Tensor, List]]:
         else:
             raise ValueError("Unknown rasterize_mode: %s", self.config.rasterize_mode)
 
-        rgb, alpha = rasterize_gaussians(  # type: ignore
-            self.xys,
-            depths,
-            self.radii,
-            conics,
-            num_tiles_hit,  # type: ignore
-            rgbs,
-            opacities,
-            H,
-            W,
-            BLOCK_WIDTH,
-            background=background,
-            return_alpha=True,
-        )  # type: ignore
-        alpha = alpha[..., None]
+        if self.config.background_color == "auto":
+            rgb, alpha = rasterize_gaussians(  # type: ignore
+                self.xys,
+                depths,
+                self.radii,
+                conics,
+                num_tiles_hit,  # type: ignore
+                rgbs,
+                opacities,
+                H,
+                W,
+                BLOCK_WIDTH,
+                background=torch.zeros(3, device=self.device),
+                return_alpha=True,
+            )  # type: ignore
+            # alpha blend rgb with background
+            alpha = alpha[..., None]
+            rgb = alpha * rgb + (1 - alpha) * (torch.sigmoid(self.learnable_bg).to(self.device) * 1.1).repeat(H, W, 1)
```

Comment:
Also I think you no longer need the extra `alpha * rgb` here, I believe. When you call rasterize_gaussians() the output rgb is already alpha blended. 

But again, I believe the proper implementation is to implement grad for background in rasterize_gaussians()